constructor/shar.py

```diff
@@ -56,70 +55,60 @@ def get_header(conda_exec, tarball, info):
         raise Exception(
             "It is not possible to use both the 'batch_mode' and "
             "'license_file' options together."
         )
     for key in 'pre_install', 'post_install', 'pre_uninstall':
-        ppd['has_%s' % key] = bool(key in info)
+        variables['has_%s' % key] = bool(key in info)
         if key in info:
-            ppd['direct_execute_%s' % key] = has_shebang(info[key])
-    ppd['initialize_conda'] = info.get('initialize_conda', True)
-    ppd['initialize_by_default'] = info.get('initialize_by_default', False)
-    ppd['has_conda'] = info['_has_conda']
-    ppd['enable_shortcuts'] = str(info['_enable_shortcuts']).lower()
-    ppd['check_path_spaces'] = info.get("check_path_spaces", True)
+            variables['direct_execute_%s' % key] = has_shebang(info[key])
+    variables['initialize_conda'] = info.get('initialize_conda', True)
+    variables['initialize_by_default'] = info.get('initialize_by_default', False)
+    variables['has_conda'] = info['_has_conda']
+    variables['enable_shortcuts'] = str(info['_enable_shortcuts']).lower()
+    variables['check_path_spaces'] = info.get("check_path_spaces", True)
     install_lines = list(add_condarc(info))
     # Omit __osx and __glibc because those are tested with shell code direcly
     virtual_specs = [
         spec
         for spec in info.get("virtual_specs", ())
         if "__osx" not in spec and "__glibc" not in spec
     ]
-    # Needs to happen first -- can be templated
-    replace = {
-        'CONSTRUCTOR_VERSION': info['CONSTRUCTOR_VERSION'],
-        'NAME': name,
-        'name': name.lower(),
-        'VERSION': info['version'],
-        'PLAT': info['_platform'],
-        'DEFAULT_PREFIX': info.get('default_prefix',
-                                   '${HOME:-/opt}/%s' % name.lower()),
-        'MD5': hash_files([conda_exec, tarball]),
-        'FIRST_PAYLOAD_SIZE': str(getsize(conda_exec)),
-        'SECOND_PAYLOAD_SIZE': str(getsize(tarball)),
-        'INSTALL_COMMANDS': '\n'.join(install_lines),
-        'CHANNELS': ','.join(get_final_channels(info)),
-        'CONCLUSION_TEXT': info.get("conclusion_text", "installation finished."),
-        'pycache': '__pycache__',
-        'SHORTCUTS': shortcuts_flags(info),
-        'REGISTER_ENVS': str(info.get("register_envs", True)).lower(),
-        'TOTAL_INSTALLATION_SIZE_KB': str(approx_size_kb(info, "total")),
-        'VIRTUAL_SPECS': shlex.join(virtual_specs),
-        'NO_RCS_ARG': info.get('_ignore_condarcs_arg', ''),
-    }
+    variables['installer_name'] = name
+    variables['installer_version'] = info['version']
+    variables['installer_platform'] = info['_platform']
+    variables['installer_md5'] = hash_files([conda_exec, tarball])
+    variables['default_prefix'] = info.get('default_prefix', '${HOME:-/opt}/%s' % name.lower())
+    variables['first_payload_size'] = str(getsize(conda_exec))
+    variables['second_payload_size'] = str(getsize(tarball))
```

Comment:
Isn't the string conversion done by jinja?

Reply:
```suggestion
    variables['first_payload_size'] = getsize(conda_exec)
    variables['second_payload_size'] = getsize(tarball)
```